### DISCUSSION TITLE
Upload to SD failed using M28/M29

### DISCUSSION THREAD

**Discussion by 4kardes** [luc-github/ESP3D#830] [General]
when i want to upload I get this error.
Change file name to short one and extension  "*.gco"
start upload but cant finish
see uploading ****.gco %100
but then
Error
    No connection

**Comment by luc-github** [comment]
what name did you use?
are you sure your  os do not hide extension ?

**Comment by 4kardes** [comment]
UI: 2.1.1 / FW: 2.1.0.c26 using Esp-01s on TFT35
example   kup.gco
I am sure again my os show extension.

**Reply by luc-github** [reply]
so issue is not linked to name but to transfer failed

**Reply by luc-github** [reply]
wait you mean : example `kup.gco` not `example kup.gco`

**Comment by 4kardes** [comment]
eample name "kup.gco"
If I type a long name, it gives an "8.3" error.
Even if I shorten the name and make the extension "gco"
transfer starts showing as "%". Even if it says "100%", it gives an error after a while. "no connection"

**Reply by luc-github** [reply]
if transfer start it means it is not name issue anymore, or it will show again the name is invalid need to change
The file transfer is very sensitive to external commands and it is often broken by TFT FW or Marlin FW 
be sure as you use esp01 on btt TFT:
* your TFT has latest firmware version
* you have a SD in your printer board sd card reader (not the TFT one)

**Comment by 4kardes** [comment]
tft firmware "BIGTREE_TFT35_V3.0_E3.27.x.bin" latest one
only printer board sd card reader have a "Upload" option. and attached.
tft sd attached but no have "upload" option (there is no button)

**Reply by luc-github** [reply]
I see this https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/pull/2431#issue-1160629826
May be this one break something as I did not tested for a while (+1 year) because I have stoped to support this feature, people try it once and complain it is slow like hell 0.4KB/s so in the end it is no more used 
So I have stop to support it

I have got some other user reported the same issue (https://github.com/luc-github/ESP3D/issues/826) - but as I cannot duplicate I do not know what is wrong and user dropped  before we debug once he learn the speed is so slow

if you can debug and tell me why it stop I may have a look on a fix but no garanty

**Comment by 4kardes** [comment]
I used his repositories
this [digant73/BIGTREETECH-TouchScreenFirmware: support TFT35 V1.0/V1.1/V1.2/V2.0/V3.0, TFT28, TFT24 V1.1, TFT43, TFT50, TFT70](https://github.com/digant73/BIGTREETECH-TouchScreenFirmware)
and
this
https://github.com/digant73/ESP3D

but nothing change still same
I need this for small laser engraving gcode files

**Comment by digant73** [comment]
I think it's a connection error due to wifi. TFT fw was ok

**Reply by 4kardes** [reply]
I just bought skr mini e3 v3 and tft35 v3.0.1+esp-01s.
what should I do. What should I check? Do you have a suggestion for me? as processing steps

**Reply by 4kardes** [reply]
I can manage all access on my printer on webui. (controlling temp, movement, all M command)
except upload. All need to send small file to sd card

**Reply by 4kardes** [reply]
Any suggestion?

**Reply by luc-github** [reply]
As told you you need to see if you can debug it
in addition because the process is sensitve to any auto report not disabled during upload, check if anything is doing query when upload is ongoing
You also can check if file size like 100K are still stuck or not like the previous post I have linked

**Comment by 4kardes** [comment]
Thanks  will try

**Comment by 4kardes** [comment]
8 kb gco file but 

same transfer starts showing as "%". Even if it says "100%", it gives an error after a while. "no connection" on TFT "sd upload failed"

**Comment by 4kardes** [comment]
I can upload any file to sd card (on skr mini e3 v3) via Octoprint
even long file name and any extension. slow but enough

**Reply by luc-github** [reply]
how do you do that ? on octoprint you upload to octprint SD (Raspberry Pi)normally not printer SD

**Reply by 4kardes** [reply]
using this https://github.com/R0peE/ESP-01-Octoprint and this https://github.com/hellerbarde/OctoPrint-Network-Printing
pyhton 3.9 using
show sd (board) content and upload any files and without any limitations

**Reply by luc-github** [reply]
I do not see where it say it use Printer SD instead of PI SD - what I read is it use ESP3D TCP port to stream files From Octopi to Printer using TCP as Serial port
Octopi as no limitation because it does not upload to Printer SD, how fast is the upload ? faster than 1MB for 12min ? https://github.com/luc-github/ESP3D/discussions/833#discussioncomment-4092488

**Reply by 4kardes** [reply]
but very slow. also I use for small Laser gcode files.
will test for speed tonight

**Reply by 4kardes** [reply]
excatly 1 mb file 12 min
